src/components/ModularTable/ModularTable.tsx

```diff
@@ -171,20 +171,45 @@ function ModularTable<D extends Record<string, unknown>>({
 
   const showEmpty: boolean = !!emptyMsg && (!rows || rows.length === 0);
 
+  // Fuction returns whether table can be sorted by a specific column.
+  const isColumnSortable = (column: HeaderGroup<D>) => {
+    // Function returns whether an instance of ReactNode has text within it.
+    // Recursively goes over all children from the node and searches for text.
+    const hasText = (node: ReactNode): boolean => {
```

Comment:
I think we could potentially modify this part not to use recursion and instead simulate the call stack with a local stack/queue. However, I'm not really sure if this would be a better approach, as the code of the new approach might look less intuitive and we will call this function on the header, which we don't expect to be a deeply nested node and thus we won't really pollute the call stack with recursion anyway.

Reply:
The following implementation of `isColumnSortable` using a local queue instead of recursion might be a better approach:

```
const isColumnSortable = (column: HeaderGroup<D>) => {
    if (!column.canSort) {
      return false;
    }

    const headerNodeQueue: ReactNode[] = [column.render("Header")];
    while (headerNodeQueue.length) {
      const frontNode = headerNodeQueue.shift();
      if (isValidElement(frontNode)) {
        headerNodeQueue.push(frontNode.props.children);
      }
      if (!isValidElement(frontNode) && Array.isArray(frontNode)) {
        frontNode.forEach((node) => headerNodeQueue.push(node));
      }
      if (
        !isValidElement(frontNode) &&
        (typeof frontNode === "string" || typeof frontNode === "number") &&
        !!String(frontNode).trim()
      ) {
        return true;
      }
    }

    return false;
  };
```